src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java

```diff
@@ -1,18 +1,134 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
 import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Observation.ObservationReferenceRangeComponent;
+import org.hl7.fhir.dstu3.model.Observation.ObservationStatus;
+import org.hl7.fhir.dstu3.model.Quantity;
+import org.hl7.fhir.dstu3.model.Quantity.QuantityComparator;
+import org.hl7.fhir.dstu3.model.SimpleQuantity;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MeasurementValueComparator;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.TestStatus;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.TestStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.InvestigationSubject;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.LabResult;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.ResultReferenceRange;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
+import java.math.BigDecimal;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Component
+@AllArgsConstructor(onConstructor_ = @Autowired)
 public class ObservationMapper {
-    public List<Observation> mapToTestGroups(final Message message) {
-        return List.of(new Observation());
+    private final UUIDGenerator uuidGenerator;
+
+    public List<Observation> mapToTestGroupsAndResults(final Message message) {
+        // test groups are GIS+N blocks with SEQ segments
+        // they should have RFF+ASL:X where X is the SEQ value of the S16
+
+        // test results are GIS+N blocks without SEQ segments
+        // they should have RFF+ARL:Y where Y is the SEQ value of the test group
+
+        final InvestigationSubject subject = message.getServiceReportDetails().getSubject();
+        final List<LabResult> labResults = subject.getLabResults();
+
+        // start by assuming everything is a test result
+        return labResults.stream().map(labResult -> {
+            final var result = new Observation();
+
+            result.setId(uuidGenerator.generateUUID());
+
+            mapStatus(labResult, result);
+            mapValueQuantity(labResult, result);
+            mapCode(labResult, result);
+            mapReferenceRange(labResult, result);
+            mapComment(labResult, result);
+
+            return result;
+        })
+            .collect(Collectors.toList());
+    }
+
+    private void mapStatus(final LabResult labResult, final Observation observation) {
```

Comment:
Could we implemented a fallback value for when there's no `STS` segment in Edifact so the status property in the `Observation` resource can be present as it's required according to the FHIR spec?

Reply:
Adding a fallback of `UNKNOWN`